libsolidity/analysis/TypeChecker.cpp

```diff
@@ -526,10 +526,6 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 			m_errorReporter.typeError(6744_error, _variable.location(), "Internal or recursive type is not allowed for public state variables.");
 	}
 
-	bool isStructMemberDeclaration = dynamic_cast<StructDefinition const*>(_variable.scope()) != nullptr;
-	if (isStructMemberDeclaration)
-		return false;
-
 	if (auto referenceType = dynamic_cast<ReferenceType const*>(varType))
```

Comment:
Is there a problem in here if it is a struct member declaration?

Reply:
Yes, it's seen in the test cases below even.
Struct members have ``Storage`` location by default, so this will trigger "Type too large for storage" errors, even though the type is never used for storage.